### DISCUSSION TITLE
adding user to cluster

### DISCUSSION THREAD

**Discussion by mflynn-lanl** [microbiomedata/nmdc-edge#267] [General]
@edwins I tried adding a new user to cluster-alma8-login1. it created the user but it says it can't create the home directory:
`useradd: cannot create directory /home/awatson2` when I try `sudo su - awatson` it puts me in `/mcflynn` which is owned by my user account

**Comment by edwins** [comment]
There are a couple of things going on. First, the `/home` directory in magic castle is mounted as an nfs share. That is why the `mcflynn` home directory was mount outside of `/home`. Second, it sounds like the uid of your new user is in conflict with mcflynn. Because of this, you'll need to specify both the home directory and an unused uid for your additional system user.

To do that, you'll need to pass in a `-d /awatson2` (or wherever you want to setup the new home directory) and the `-u 1002` (or any unused uid; that was just an example). To determine what number hasn't been used, just look at the `/etc/passwd` file. Each entry in that file is a colon-delimited set of values. I suggest something between 1002 and 1999 e.g. the next number after `mcflynn`. Note, the uid is the 3rd field (first number).

**Reply by mflynn-lanl** [reply]
I added a new user with the command
`sudo useradd -G adm,wheel,systemd-journal -u 1003 -s /bin/bash -m -d /apwatson apwatson`
and that worked fine. but when I try to do `sudo su - user` I get: 
```
We trust you have received the usual lecture from the local System
Administrator. It usually boils down to these three things:

    #1) Respect the privacy of others.
    #2) Think before you type.
    #3) With great power comes great responsibility.

[sudo] password for apwatson:
```
I don't see a `sudo` group and I'm not able to add the user to group `sudo`

**Reply by mflynn-lanl** [reply]
I tried `su - user` and used the password in `accounts.txt` but that didn't work

**Reply by mflynn-lanl** [reply]
@edwins I'd like to create virtual clusters to test different versions of the website and workflow code.

**Reply by edwins** [reply]
I'll take a look at the issue.

Also, as an update, there might be a way to use manila shares (rather than volumes) as an alternative to sharing data across magic castle, web application, and any other vms you might have. If this works as I would expect, then it would help you avoid doing volume snapshots entirely. Let me work out the details, and show you how it would work.

**Reply by edwins** [reply]
It looks like other MC users are encountering similar symptoms you just reported. This was just reported today. I'll keep you informed as this develops.

![image](https://github.com/user-attachments/assets/840d9be3-2dff-494b-8214-02b29ff4c5a9)
